app/src/components/Map/SearchModal.jsx

```diff
@@ -0,0 +1,124 @@
+import CloseIcon from 'babel!svg-react!assets/icons/close.svg?name=CloseIcon';
+
+class SearchModal extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.onPageChange = this.onPageChange.bind(this);
+  }
+
+  onSearchInputChange(value) {
+    this.props.setSearchTerm(value);
+  }
+
+  onPageChange(value) {
+    this.props.setSearchPage(value.selected);
+  }
+
+  cleanResults() {
+    this.props.setSearchTerm('');
+  }
+
+  render() {
+    let searchResults;
+
+    if (this.props.searching) {
+      searchResults = <li className={SearchModalStyles['status-message']}>Searching...</li>;
+    } else if (this.props.count && this.props.searchTerm.length >= SEARCH_QUERY_MINIMUM_LIMIT) {
+      searchResults = [];
+      for (let i = 0, length = this.props.entries.length; i < length; i++) {
+        searchResults.push(<SearchResult
+          className={classnames(ResultListStyles['result-item'], SearchModalStyles['search-result-item'])}
+          key={i}
+          closeSearch={() => this.props.closeSearchModal()}
+          vesselInfo={this.props.entries[i]}
+        />);
+      }
+    } else if (this.props.searchTerm.length < SEARCH_QUERY_MINIMUM_LIMIT && this.props.searchTerm.length > 0) {
+      searchResults = (
+        <li className={SearchModalStyles['status-message']}>
+          Type at least {SEARCH_QUERY_MINIMUM_LIMIT} characters
+        </li>);
+    } else {
+      searchResults = <li className={SearchModalStyles['status-message']}>No result</li>;
+    }
+
+    return (
+      <div className={SearchModalStyles['c-search-modal']}>
+        <h3 className={ModalStyles['modal-title']}>Search vessel</h3>
+        <div className={SearchModalStyles['search-container']}>
+          <div className={SearchModalStyles['search-input-container']}>
+            <input
+              className={SearchModalStyles['search-input']}
+              onChange={(e) => this.onSearchInputChange(e.target.value)}
+              placeholder="Search vessel"
+              value={this.props.searchTerm}
+            />
+            {this.props.searchTerm.length === 0 && <SearchIcon
+              className={classnames(iconsStyles.icon, SearchModalStyles['search-icon'])}
+            />}
+            {this.props.searchTerm.length > 0 && <CloseIcon
+              className={classnames(iconsStyles.icon, iconsStyles['icon-close'], SearchModalStyles['delete-icon'])}
+              onClick={() => this.cleanResults()}
+            />}
+          </div>
+          {searchResults && <ul className={classnames(ResultListStyles['c-result-list'], SearchModalStyles['search-result-list'])}>
+            {searchResults}
+          </ul>
+          }
+        </div>
+        <div className={SearchModalStyles['paginator-container']}>
+          <div
+            className={PaginatorStyles['c-paginator']}
+            style={(!this.props.searching && this.props.entries.length > 0) ? {} : { display: 'none' }}
```

Comment:
https://github.com/AdeleD/react-paginate/issues/91

If we do not render the paginator when in "searching" state, then we need to pass it an "initialPage" parameter to set the page (otherwise always shows the first).
If we do this, it triggers the page changed callback, which basically creates an endless loop. So, we can't remove the Paginator, just hide it. A bit hacky, I know.